binary-decode.go

```diff
@@ -677,12 +677,12 @@ func (cdc *Codec) decodeReflectBinaryStruct(bz []byte, info *TypeInfo, rv reflec
 			// So in the future, this may not match,
 			// so we will need to remove this sanity check.
 			if field.BinFieldNum != fieldNum {
-				err = errors.New(fmt.Sprintf("Expected field number %v, got %v", field.BinFieldNum, fieldNum))
+				err = errors.New(fmt.Sprintf("expected field number %v, got %v", field.BinFieldNum, fieldNum))
 				return
 			}
 			typWanted := typeToTyp4(field.Type, field.FieldOptions).Typ3()
 			if typ != typWanted {
-				err = errors.New(fmt.Sprintf("Expected field type %X, got %X", typWanted, typ))
+				err = errors.New(fmt.Sprintf("expected field type %v, got %v", typWanted, typ))
```

Comment:
I noticed you replaced %x with %v in What information will this spit out now? previously the error I received was:
`Expected field type 496E74657266616365, got 3842797465`
what may the output look like now?

Reply:
Previously:
`Expected field type 3842797465, got 496E74657266616365``
now:
`Expected field type 8Byte, got Interface`

`Typ3` implements the stringer interface: 
https://github.com/tendermint/go-amino/blob/ecf42f8907e063f630fcce7605c61252db93c355/amino.go#L34-L38

example here:
https://gist.github.com/Liamsi/c8dc86fb770bce38eec45b832f511584